rogueserver_test.go

```diff
@@ -0,0 +1,285 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"github.com/pagefaultgames/rogueserver/api/account"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"io"
+	"net"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"strings"
+	"testing"
+
+	"github.com/pagefaultgames/rogueserver/api"
+	"github.com/pagefaultgames/rogueserver/db"
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/modules/mariadb"
+)
+
+func TestRogueServer(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping test in short mode")
+	}
+
+	setupDB(t)
+	apiURL := runServer(t)
+
+	username := "user1234"
+	oldPassword := "password1234"
+	newPassword := "password12345"
+	t.Run("/account", func(t *testing.T) {
+		t.Run("/account/register", func(t *testing.T) {
+			t.Run("invalid username", func(t *testing.T) {
+				resp := postHttpForm(t, apiURL, "/account/register", nil, url.Values{"username": {""}, "password": {"password"}})
+				resp.assertStringResponse(t, http.StatusInternalServerError, "invalid username\n")
+			})
+
+			t.Run("invalid password", func(t *testing.T) {
+				resp := postHttpForm(t, apiURL, "/account/register", nil, url.Values{"username": {"username"}, "password": {"1"}})
+				resp.assertStringResponse(t, http.StatusInternalServerError, "invalid password\n")
+			})
+			t.Run("register", func(t *testing.T) {
+				resp := postHttpForm(t, apiURL, "/account/register", nil, url.Values{"username": {username}, "password": {oldPassword}})
+				resp.assertStringResponse(t, http.StatusOK, "")
+			})
+			t.Run("fail when register called for already registered user", func(t *testing.T) {
+				resp := postHttpForm(t, apiURL, "/account/register", nil, url.Values{"username": {username}, "password": {oldPassword}})
+				resp.assertStringResponse(t, http.StatusInternalServerError, "failed to add account record: Error 1062 (23000): Duplicate entry 'user1234' for key 'username'\n")
+			})
+
+		})
+
+		var token string
+		t.Run("/account/login", func(t *testing.T) {
+			t.Run("invalid username", func(t *testing.T) {
+				resp := postHttpForm(t, apiURL, "/account/login", nil, url.Values{"username": {""}, "password": {"password"}})
+				resp.assertStringResponse(t, http.StatusInternalServerError, "invalid username\n")
```

Comment:
Invalid request data should not prompt a HTTP 5xx response.

Can you make this test for a HTTP 400 (Bad Request) status instead?